app/controllers/income/estimatedPay/update/IncomeUpdateCalculatorController.scala

```diff
@@ -136,52 +128,49 @@ class IncomeUpdateCalculatorController @Inject() (
+                    Future
+                      .successful(Redirect(routes.IncomeUpdateEstimatedPayController.estimatedPayLandingPage(empId)))
                   case Some(FormValuesConstants.NoValue) =>
-                    Redirect(controllers.routes.IncomeSourceSummaryController.onPageLoad(empId))
+                    Future.successful(Redirect(controllers.routes.IncomeSourceSummaryController.onPageLoad(empId)))
                 }
             )
-        }
+        case _ => Future.successful(errorPagesHandler.internalServerError("Mandatory values missing"))
+      }
   }
 
   // scalastyle:off method.length
-  def checkYourAnswersPage(empId: Int): Action[AnyContent] = authenticate.authWithValidatePerson.async {
+  def checkYourAnswersPage(empId: Int): Action[AnyContent] = authenticate.authWithDataRetrieval.async {
     implicit request =>
       implicit val user: AuthedUser = request.taiUser
-
-      val collectedValues = journeyCacheService
-        .collectedJourneyValues(
-          Seq(
-            UpdateIncomeConstants.NameKey,
-            UpdateIncomeConstants.PayPeriodKey,
-            UpdateIncomeConstants.TotalSalaryKey,
-            UpdateIncomeConstants.PayslipDeductionsKey,
-            UpdateIncomeConstants.BonusPaymentsKey,
-            UpdateIncomeConstants.IdKey
-          ),
-          Seq(
-            UpdateIncomeConstants.TaxablePayKey,
-            UpdateIncomeConstants.BonusOvertimeAmountKey,
-            UpdateIncomeConstants.OtherInDaysKey
-          )
-        )
-
-      collectedValues.map {
-        case Left(errorMessage) =>
-          logger.warn(errorMessage)
-          Redirect(controllers.routes.IncomeSourceSummaryController.onPageLoad(empId).url)
-        case Right((mandatorySeq, optionalSeq)) =>
-          val employer = IncomeSource(id = mandatorySeq(5).toInt, name = mandatorySeq.head)
-          val payPeriodFrequency = mandatorySeq(1)
-          val totalSalaryAmount = mandatorySeq(2)
-          val hasPayslipDeductions = mandatorySeq(3)
-          val hasBonusPayments = mandatorySeq(4)
+      val userAnswers = request.userAnswers
+
+      val mandatoryJourneyValues = Seq(
+        userAnswers.get(UpdateIncomeNamePage),
+        userAnswers.get(UpdateIncomePayPeriodPage),
+        userAnswers.get(UpdateIncomeTotalSalaryPage),
+        userAnswers.get(UpdateIncomePayslipDeductionsPage),
+        userAnswers.get(UpdateIncomeBonusPaymentsPage),
+        userAnswers.get(UpdateIncomeIdPage)
+      )
+      val optionalSeq = Seq(
+        userAnswers.get(UpdateIncomeTaxablePayPage),
+        userAnswers.get(UpdateIncomeBonusOvertimeAmountPage),
+        userAnswers.get(UpdateIncomeOtherInDaysPage)
+      )
+
+      (mandatoryJourneyValues, optionalSeq) match {
+        case (mandatory, _) if mandatory.forall(_.isDefined) =>
+          val employer =
+            IncomeSource(id = mandatory(5).getOrElse("").toString.toInt, name = mandatory.head.getOrElse("").toString)
```

Comment:
Dont need getOrElse as you have already confirmed it is defined

Reply:
Addressed